src/linear_algebra/basic.lean

```diff
@@ -1042,6 +1044,21 @@ end
 @[reducible] def range_restrict (f : M →ₗ[R] M₂) : M →ₗ[R] f.range :=
 f.cod_restrict f.range f.mem_range_self
 
+section
+variables (R) (M)
+
+/-- Given an element `x` of a module `M` over `R`, the natural map from
+    `R` to scalar multiples of `x`.-/
+def span_singleton (x : M) : R →ₗ[R] M := linear_map.id.smul_right x
```

Comment:
This name feels slightly wrongish... it doesn't sound enough like a map to me. But I don't have an obvious suggestion, beyond prepending `to_` which makes it longer...

Reply:
I don't love the name either.  Happy to change to `to_` if you prefer, or maybe someone else has an idea.

Reply:
`to_span_singleton` sounds good to me.

Reply:
Changed in 07bf973.

Reply:
By the way, an alternative is just to use `id.smul_right` throughout, for all these lemmas, instead of renaming `id.smul_right` as `to_span_singleton`.